src/puppetlabs/trapperkeeper/services/scheduler/scheduler_core.clj

```diff
@@ -136,6 +136,17 @@
       ; this can occur if the interface is being used while the scheduler is shutdown
       (log/error e (i18n/trs "Failed to schedule job")))))
 
+(defn cron-next-valid-time
+  "Returns the next occurance of the cron specification after the given date"
+  [cron-string ^Date date]
+  (let [cron (CronExpression. cron-string)
+        trigger (.withSchedule (TriggerBuilder/newTrigger)
+                               (.startAtDate (DateBuilder/dateOf (.getYear date) (.getMonth date) (.getDate date)))
```

Comment:
I'm surprised you need the DateBuilder here, as you already have a `Date` object.  Can you just use that directly?